tests/unit/test_api.py

```diff
@@ -204,6 +201,50 @@ class TestCaseDict(TypedDict):
                 self.assertEqual(base_urls.is_sonar_qube_cloud, expected.is_sonar_qube_cloud)
                 self.assertEqual(base_urls, expected)
 
+    def test_inconsistent_urls_raises_exception(self):
```

Comment:
I am not super fan of the nested `with`s here, I find it a bit more difficult to read. Still it is a preference, but I would rather go with a loop over a dict with which error is raised depending on which config is passed, much like `test_get_base_urls`. but I don't expect any changes.

Reply:
Good point! For the record, I actually don't really like these `with` sub-test syntax, as I feel the feedback from failing test is not as good as with proper tests. However, I refactored to make sure it stays consistent.